src/modules/trajectory_generation/domain/spline_generation/cubic_spline_generator.hpp

```diff
@@ -2,19 +2,160 @@
+             * 
+             * @param p1 
+             * @param p2 
+             * @param s 
+             * @return std::vector<double> the spline parameters {a, b, c, d}
+             */
+            std::vector<double> calculate_parameters(double p1, double p2, double s);
+
+            /**
+             * @brief Get the next state object
+             * 
+             * @param params 
+             * @param s 
+             * @return core_datastructures::Posture return next state given spline params
+             */
+            core_datastructures::Posture get_next_state(std::vector<double>& params, double s);
+
+
+            /**
+             * @brief Computes Jacobian matrix for the given params
+             * 
+             * @param params 
+             * @return Eigen::Matrix3d Jacobian Matrix
+             */
+            Eigen::Matrix3d calculate_Jacobian(std::vector<double>& params);
+
+
+            /**
+             * @brief Runs graident descent to compute final spline parameters
+             * 
+             */
+            void run_gradient_descent();
+
+            /**
+             * @brief Outputs a set of postures after computing the final spline params
+             * 
+             * @return std::vector<core_datastructures::Posture> 
+             */
+            std::vector<core_datastructures::Posture> generate_splines();
+
+            /**
+             * @brief Computes curvature given params and arc length
+             * 
+             * @param params 
+             * @param s 
+             * @return double Curvature
+             */
+            double calculate_kappa(std::vector<double>& params, double s);
+
+            /**
+             * @brief Computes theta given params and arc length
+             * 
+             * @param params 
+             * @param s 
+             * @return double theta value in radians
+             */
+            double calculate_theta(std::vector<double>& params, double s);
+
+            // double calculate_x();
+            /**
+             * @brief computes x given params and arc length
+             * 
+             * @param params 
+             * @param s 
+             * @return double 
+             */
+            double calculate_x(std::vector<double>& params, double s);
+
+            /**
+             * @brief Computes y given params and arc length
+             * 
+             * @param params 
+             * @param s 
+             * @return double 
+             */
+            double calculate_y(std::vector<double>& params, double s);
+
+            /**
+             * @brief Gets the euclidian distance given two postures
+             * 
+             * @param start 
+             * @param goal 
+             * @return double eucilidian distance
+             */
+            double get_distance(const core_datastructures::Posture& start, const core_datastructures::Posture& goal);
```

Comment:
Since this function does not depend on any of the spline features, maybe we can shift it to another file named spline_utils.cpp and spline_utils.hpp. If it is too much work, feel free to resolve. 

Same goes for comp() function and get_delta() function